packages/graphql-toolbox/src/modules/SchemaView/SchemaSettings.tsx

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react";
+import { Checkbox, HeroIcon, Radio } from "@neo4j-ndl/react";
+import { ProTooltip } from "../../components/ProTooltip";
+import { Storage } from "src/utils/storage";
+import {
+    LOCAL_STATE_CHECK_CONSTRAINT,
+    LOCAL_STATE_CREATE_CONSTRAINT,
+    LOCAL_STATE_ENABLE_DEBUG,
+    LOCAL_STATE_ENABLE_REGEX,
+} from "src/constants";
+
+interface Props {
+    isRegexChecked: string | null;
+    isDebugChecked: string | null;
+    isCheckConstraintChecked: string | null;
+    isCreateConstraintChecked: string | null;
+    setIsRegexChecked: React.Dispatch<React.SetStateAction<string | null>>;
+    setIsDebugChecked: React.Dispatch<React.SetStateAction<string | null>>;
+    setIsCheckConstraintChecked: React.Dispatch<React.SetStateAction<string | null>>;
+    setIsCreateConstraintChecked: React.Dispatch<React.SetStateAction<string | null>>;
+}
+
+export const SchemaSettings = ({
+    isRegexChecked,
+    isDebugChecked,
+    isCheckConstraintChecked,
+    isCreateConstraintChecked,
+    setIsRegexChecked,
+    setIsDebugChecked,
+    setIsCheckConstraintChecked,
+    setIsCreateConstraintChecked,
+}: Props) => {
+    const onChangeRegexCheckbox = (): void => {
+        const next = isRegexChecked === "true" ? "false" : "true";
```

Comment:
**suggestion, non-blocking**: I understand that these are states for react actions, maybe we could rename them to avoid confusion with booleans? e.g. `"checked", "unchecked"`

Reply:
Yes I see your point! However I prefer the boolean values over `checked` and `unchecked` as they are a bit more general and not directly linked to the "terminology" of a checkbox state. For instance if we have to refactor the UI to represent it in some other UI element, a boolean is more flexible but still descriptive.